src/copilot/tools/listViewsTool.ts

```diff
@@ -0,0 +1,111 @@
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as vscode from "vscode";
+import { ToolBase } from "./toolBase";
+import ConnectionManager from "../../controllers/connectionManager";
+import * as Constants from "../../constants/constants";
+import { MssqlChatAgent as loc } from "../../constants/locConstants";
+import { getErrorMessage } from "../../utils/utils";
+import SqlToolsServiceClient from "../../languageservice/serviceclient";
+import { RequestType } from "vscode-languageclient";
+import { SimpleExecuteResult } from "../interfaces";
+import { listViewsQuery } from "../queries";
+
+export interface ListViewsToolParams {
+    connectionId: string;
+}
+
+export interface ListViewsToolResult {
+    success: boolean;
+    message?: string;
+    views: string[];
+}
+
+export class ListViewsTool extends ToolBase<ListViewsToolParams> {
+    public readonly toolName = Constants.copilotListViewsToolName;
+
+    constructor(
+        private _connectionManager: ConnectionManager,
+        private _client: SqlToolsServiceClient,
+    ) {
+        super();
+    }
+
+    async call(
+        options: vscode.LanguageModelToolInvocationOptions<ListViewsToolParams>,
+        _token: vscode.CancellationToken,
+    ) {
+        const { connectionId } = options.input;
+        try {
+            const connInfo = this._connectionManager.getConnectionInfo(connectionId);
+            const connCreds = connInfo?.credentials;
+            if (!connCreds) {
+                return JSON.stringify({
+                    success: false,
+                    message: loc.noConnectionError(connectionId),
+                });
+            }
+
+            const result = await this._client.sendRequest(
+                new RequestType<
+                    { ownerUri: string; queryString: string },
+                    SimpleExecuteResult,
+                    void,
+                    void
+                >("query/simpleexecute"),
+                {
+                    ownerUri: connectionId,
+                    queryString: listViewsQuery,
+                },
+            );
+            const views = this.getViewNamesFromResult(result);
+
+            return JSON.stringify({
+                success: true,
+                views,
+            });
+        } catch (err) {
+            return JSON.stringify({
+                success: false,
+                message: getErrorMessage(err),
+            });
+        }
+    }
+
+    private getViewNamesFromResult(result: SimpleExecuteResult): string[] {
+        if (!result || !result.rows || result.rows.length === 0) {
+            return [];
+        }
+
+        const viewNames: string[] = [];
+
+        // Extract view names from each row
+        // Assuming the query returns view names in the first column
```

Comment:
Add a disclaimer for this one too to add additional columns to the end of the column list to prevent breaking the assumption mentioned here.